plugins/pipelines/metrics.js

```diff
@@ -24,24 +25,33 @@ module.exports = () => ({
         handler: (request, reply) => {
             const factory = request.server.app.pipelineFactory;
             const { id } = request.params;
-            const { aggregateInterval } = request.query;
+            const { aggregateInterval, page, count } = request.query;
             let { startTime, endTime } = request.query;
 
-            if (!startTime || !endTime) {
-                ({ startTime, endTime } = setDefaultTimeRange(startTime, endTime, MAX_DAYS));
-            }
-
             return factory.get(id)
                 .then((pipeline) => {
                     if (!pipeline) {
                         throw boom.notFound('Pipeline does not exist');
                     }
 
-                    if (!validTimeRange(startTime, endTime, MAX_DAYS)) {
-                        throw boom.badRequest(`Time range is longer than ${MAX_DAYS} days`);
-                    }
+                    let config;
 
-                    const config = { startTime, endTime };
+                    // Only when either page or count is unavailable
+                    // check whether startTime and endTime are valid
+                    if (!page || !count) {
```

Comment:
Shouldn't this be if (!page && !count) (https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L172-L178)
Default count is 50, so if only page is set, it will use count = 50
Default page = 1, so if only count is set, it will use page = 1

If BOTH are not available then it will use startTime & endTime

Reply:
I think we only allow fetching by pagination when both page and count are available in the config?

Reply:
well model implementation needs both https://github.com/screwdriver-cd/models/pull/399/files#diff-679e90b5b15ab1b73ddb7ed2377e7413R1295
unless that's changed this change here is good.

Reply:
Why do we suddenly require to have BOTH fields? It has always been you only need `page` or `count` (https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L172-L179). The other will use default value. Why should the behavior be different here?

Reply:
well now we need to start changing from model. This should have been caught earlier in review.

Reply:
@Code-Beast could you please handle this scenario

Reply:
Sorry did not review the previous PR. We can merge this one and fix afterward. It's not too critical, but we should fix it so that the behavior is consistent.

Reply:
sure